src/ui/public/share/directives/share_object_url.js

```diff
@@ -69,8 +70,14 @@ app.directive('shareObjectUrl', function (Private, Notifier) {
         });
       };
 
+      // since getUrl() is called within a watcher we cache the unhashing step
+      const unhashStatesInAbsUrl = memoize((absUrl) => {
```

Comment:
That and the visual of an auto-incrementing counter in the URL seems strange to me.